skimage/feature/fisher_vector.py

```diff
@@ -244,19 +244,25 @@ def fisher_vector(descriptors, gmm, *, improved=False, alpha=0.5):
     # Compute GMM gradients wrt its parameters
     d_pi = pp_sum.squeeze() - mixture_weights
 
-    d_mu = (pp_x - pp_sum * means).ravel()
+    d_mu = pp_x - pp_sum * means
 
     d_sigma_t1 = pp_sum * np.power(means, 2)
     d_sigma_t2 = pp_sum * covariances
     d_sigma_t3 = 2 * pp_x * means
-    d_sigma = (-pp_x_2 - d_sigma_t1 + d_sigma_t2 + d_sigma_t3).ravel()
+    d_sigma = -pp_x_2 - d_sigma_t1 + d_sigma_t2 + d_sigma_t3
+
+    # Apply analytical diagonal normalization
+    sqrt_mixture_weights = np.sqrt(mixture_weights)
+    d_pi /= sqrt_mixture_weights
+    d_mu /= np.expand_dims(sqrt_mixture_weights, -1) * covariances
+    d_sigma /= np.expand_dims(np.sqrt(2 * mixture_weights), -1) * covariances ** 2
```

Comment:
Thanks. This normalization seems to have broken the performance in the example script. The macro avg accuracy decreased from ~0.72 to ~0.17. 

I think there is an error in that d_mu should be using `sqrt(covariances)` rather than `covariances`
and d_sigma should be using `covariances` instead of `covariances ** 2`.
If I make those changes the accuracy in the demo improves, but only to ~0.63 which is still lower than what it was before.

Reply:
```suggestion
    sqrt_mixture_weights = np.sqrt(mixture_weights)
    d_pi /= sqrt_mixture_weights
    d_mu /= sqrt_mixture_weights[:, np.newaxis] * np.sqrt(covariances)
    d_sigma /= np.sqrt(2) * sqrt_mixture_weights[:, np.newaxis] * covariances
```